### PR TITLE
New improved implementation of bubble sort

### PR DESCRIPTION
**Pull request by shanshaji**
The current implementation is iterating/looping the length of the array inside the loop. The bubble sort is called bubble sort because in each pass-through, the highest unsorted value “bubbles” up to its correct position. So that those elements can be omitted in the upcoming loop. This new implementation wouldn't go through the whole array every time. Once an element is sorted, it's not considered in the next loop

### CODE REVIEW

**Review by vzvu3k6k — CHANGES_REQUESTED** [review]
Thanks @shanshaji.

Can you remove IDE config files (`.idea/*`) from this PR? I think they are irrelevant with the subject of this PR.

File: sorting/bubble_sort.rb [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+  until sorted
+    sorted = true
+    for i in (0..unsorted_until_index)
+      if array[i+1]
+        if array[i] > array[i+1]
+          array[i], array[i+1] = array[i+1], array[i]
+          sorted = false
+        end
       end
     end
```
Comment (vzvu3k6k) [inline sorting/bubble_sort.rb:15]: Suggestion: I think [an index check](https://github.com/TheAlgorithms/Ruby/pull/186/files#diff-0ca68e1c2f3cb3b62610801c9f5751b1c10225212263365927998e84c71b15e0R9) can be removed with [exclusive range literal](https://ruby-doc.org/core-3.0.3/Range.html#:~:text=those%20created%20using%20...%20exclude%20the%20end%20value.).

```suggestion
    for i in (0...unsorted_until_index)
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        sorted = false
      end
    end
```

---

Another way:

```suggestion
    0.upto(unsorted_until_index - 1) do |i|
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        sorted = false
      end
    end
```

Comment (shanshaji) [reply sorting/bubble_sort.rb]: I have changed the code according to the suggestion

**Review by vzvu3k6k — APPROVED** [review]
Thanks! Looks good.

@vbrazo Can you merge this PR?

**Review by vbrazo — APPROVED** [review]

**Pull request merged**